server/src/main/java/org/elasticsearch/search/SearchFeatures.java

```diff
@@ -16,8 +16,11 @@
 import java.util.Set;
 
 public final class SearchFeatures implements FeatureSpecification {
+
+    public static final NodeFeature LUCENE_10_0_0_UPGRADE = new NodeFeature("lucene_10_upgrade");
```

Comment:
Turns out @benwtrent had a very good intuition. The day after we chatted about it, i had to introduce this to address the profile yaml test failures. See https://github.com/elastic/elasticsearch/commit/c0646368f782decdd9d1109c25e12b36746c6e1a . Heads up this node feature is already available in the branch.

Reply:
Removed from the diff now.